### PR TITLE
feat(curation): add service_id

### PR DESCRIPTION
**Pull request by linkeralbertlo**
## Purpose

same as title

- [AB#20112](https://dev.azure.com/linkerengineer/87361b6b-4b8a-4d65-8243-96cf1163a72f/_workitems/edit/20112)
- [AB#20114](https://dev.azure.com/linkerengineer/87361b6b-4b8a-4d65-8243-96cf1163a72f/_workitems/edit/20114)
- [AB#20137](https://dev.azure.com/linkerengineer/87361b6b-4b8a-4d65-8243-96cf1163a72f/_workitems/edit/20137)

## What Changes?

- support upload visionai format, image format directly from local
- implement async upload 

## What to Check?

- upload from local

```python3
from dataverse_sdk import DataverseClient, DataverseHost
from dataverse_sdk.schemas.client import Project
from dataverse_sdk.schemas.common import AnnotationFormat, DatasetType, DataSource

client = DataverseClient(
    host=DataverseHost.LOCAL,
    email="bc51f3c57a5e9cdbb8a74614@local.email.com",
    password="Pass1234",
    service_id="ed47b043-90c0-4272-afc9-f60041c612e6",
)

p: Project = client.get_project(1)
p.create_dataset(
    name="coco4",
    data_source=DataSource.SDK,
    sensors=p.sensors,
    type=DatasetType.RAW_DATA,
    annotation_format=AnnotationFormat.VISION_AI,
    storage_url="",
    data_folder="/Users/albertlo/Desktop/coco_to_visionai",
    container_name="",
    sas_token="",
    access_key_id="",
    secret_access_key="",
)
print(p.dict())

```

截圖 2024-05-17 下午4 33 55

### CODE REVIEW

File: python/dataverse_sdk/client.py [hunk 1]
```diff
@@ -891,60 +894,162 @@ def create_dataset(
...
+    ) -> tuple[deque, str, list[str]]:
+        max_retry_count, batch_size = 3, 50
+
+        failed_urls = []
+        upload_task_queue = deque()
+
+        # TODO: convert the following code to async tasks loop
+        generate_url_queue = deque()
+        for i in range(0, len(file_paths), batch_size):
+            generate_url_queue.append((file_paths[i : i + batch_size], 0))
```
Comment (henrychoulinker) [inline python/dataverse_sdk/client.py:942]: 這邊會有資料錯或是out index的風險

File: python/setup.py [hunk 2]
```diff
@@ -11,11 +11,11 @@
     name=PACKAGE_NAME,
     version=PACKAGE_VERSION,
     packages=find_packages(),
-    python_requires=">=3.9, <4",
+    python_requires=">=3.10, <4",
     author=AUTHOR,
     url="",
     description=DESC,
-    install_requires=["pydantic==1.*", "requests"],
+    install_requires=["pydantic==1.*", "requests", "httpx==0.23.0"],
```
Comment (kyolt) [inline python/setup.py:18]: httpx 一定要 0.23.0 嗎，給個 range?

File: python/README.md [hunk 3]
```diff
@@ -365,4 +365,4 @@ status, onnx_model_path = model.get_onnx_model_file(save_path="./model.onnx", ti
 
 ## Links to language repos
 
-[Python Readme](https://github.com/linkernetworks/dataverse-sdk/tree/develop/python/README.md)
+[Python Readme](https://github.com/linkernetworks/dataverse-sdk/tree/master/python/README.md)
```
Comment (yihsuan123) [inline python/README.md:368]: Might need to update README for the connection to Dataverse with service_id

File: python/dataverse_sdk/client.py (hunk 1 above)
Comment (henrychoulinker) [reply python/dataverse_sdk/client.py:942]: image
thx

Comment (linkeralbertlo) [reply python/dataverse_sdk/client.py:942]: addressed in person

File: python/setup.py (hunk 2 above)
Comment (linkeralbertlo) [reply python/setup.py]: >= 0.23.0

**Review by henrychoulinker — APPROVED** [review]

**Review by yihsuan123 — APPROVED** [review]

**Review by kyolt — APPROVED** [review]

**Pull request merged**